rust/src/io/deletion.rs

```diff
@@ -341,6 +343,74 @@ pub(crate) async fn read_deletion_file(
     }
 }
 
+pub struct LruDeletionVectorStore {
```

Comment:
This looks good. In the future, we might have the reads make concurrent queries on `is_deleted()`, in which case we might change this to:

1. Use Rwlock instead of Mutex (unless we find writer starvation is an issue)
2. Make the IO happen separately, to prevent the situation where two concurrent calls are making the same IO call.

Reply:
agreed. I tried to use `RWLock` but `LruCache` requires mut ref for getting items too (updating the recently used state probably)

I need to find a way to go around that